File: res/JavaSourceSample.py
class JavaSourceSample:
  crossVariable = ''
  obsFactor = 9

  def __init__(self, lazyIdentity=0):
    self.lazyIdentity = lazyIdentity

  def instanceFun(self):
    print(self.lazyIdentity)

  def staticFun():
    print(JavaSourceSample.obsFactor)

  def read():
    data = input()
    print(data)

  def show(self, j, smthg, o):
    x = 0
    d = 4.1
    b = 10.1
    s = "cadena"
    n = 9292.2
    print(d)
    print(s)
    print(n)

  def ifStatement(self):
    temperatura = 24
    condicion1 = True
    condicion2 = False
    if (temperatura>25):
      print("A la playa!!!")
    if (temperatura<=25):
      print("Esperando al buen tiempo...")
    if (temperatura>25):
      print("A la playa!!!")
    else:
      print("Esperando al buen tiempo...")
    if (condicion1):
      print("A")
    else:
      if (condicion2):
        print("B")
      else:
        if (condicion2):
            if (temperatura>25):
              print("A la playa!!!")
            else:
              print("Esperando al buen tiempo...")
        else:
          print("D")

  def switchExample(self):
    day = 5
    day2 = 2
    dayString = ''
    dayString2 = ''
    if (day == 1):
      if (day == 1):
        dayString="Lunes"
      elif (day == 2):
        dayString="2"
      else:
        dayString="3"
    elif (day == 2):
      dayString="Martes"
    elif (day == 3):
      dayString="Miercoles"
    elif (day == 4):
      dayString="Jueves"
    elif (day == 5):
      dayString="Viernes"
    elif (day == 6):
      dayString="Sabado"
    elif (day == 7):
      dayString="Domingo"
    else:
      dayString="Dia inválido"
    if (day2 == 1):
      dayString2="1"
    elif (day2 == 2):
      dayString2="2"
    else:
      dayString2="3"
    print(dayString)
    print(dayString2)

  def whileLoop(self):
    print("Test para 'while':")
    i = 2
    while i<10:
      print(i)
      self.forLoop()
      i = i+1

  def doWhileLoop(self):
    print("Test para 'doWhile':")
    i = 1
    while True:
      print(i)
      i = i+1
      if not i<=3:
        break

  def forLoop(self):
    i = 2
    while i<500 and i%1==0:
      print(i/2)
      i=i*i
      
    j = 12
    j = 2
    while not (j>30):
      print(j)
      j=j*j
      
    k = 2
    while k<10:
      pass
      k=k*k
      

  def main(args):
    print("Hello world!")
    j = JavaSourceSample()
    j.show(3, "algo", args)
    print("------------------------------")
    j.forLoop()
    print("..............................")
    j.instanceFun()
    print("------------------------------")
    j.whileLoop()
    print("..............................")
    JavaSourceSample.staticFun()
    print("------------------------------")
    j.switchExample()
    print("..............................")
    j.ifStatement()
    print("------------------------------")
    j.doWhileLoop()
    print("..............................")
    JavaSourceSample.read()

import sys
if __name__ == '__main__':
  JavaSourceSample.main(sys.argv)

File: in/JavaSourceSample.java
import java.util.Scanner;

public class JavaSourceSample {

    public static String crossVariable;
    public int lazyIdentity = 0;
    public static double obsFactor = 9;

    public void instanceFun() {
        System.out.println(lazyIdentity);
    }

    public static void staticFun() {
        System.out.println(obsFactor);
    }

    public static void read() {
        Scanner input = new Scanner(System.in);
        String data = input.nextLine();
        System.out.println(data);
    }

    public String show(Integer j, String smthg, Object o) {
        Integer x;
        Double d = 4.1, b = 10.1;
        String s = "cadena";
        double n = 9292.2;
        System.out.println(d);
        System.out.println(s);
        System.out.println(n);
        return "";
    }

    public void ifStatement() {
        int temperatura = 24;
        boolean condicion1 = true;
        boolean condicion2 = false;

        if (temperatura > 25) {
            System.out.println("A la playa!!!");
        }

        if (temperatura <= 25) {
            System.out.println("Esperando al buen tiempo...");
        }
        if (temperatura > 25) {
            System.out.println("A la playa!!!");
        } else {
            System.out.println("Esperando al buen tiempo...");
        }

        if (condicion1==true) {
            System.out.println("A");
        } else if (condicion2==false) {
            System.out.println("B");
        } else if (condicion2) {
            if (temperatura > 25) {
                System.out.println("A la playa!!!");
            } else {
                System.out.println("Esperando al buen tiempo...");
            }
        } else {
            System.out.println("D");
        }
    }

    public void whileLoop() {
        System.out.println("Test para 'while':");
        int i = 2;
        boolean b = true;
        while (i < 10) {
            System.out.println(i);
            forLoop();
            i = i + 1;
        }
        while (b) {
            System.out.println(i);
            forLoop();
            b = false;
        }
    }

    public void doWhileLoop() {
        System.out.println("Test para 'doWhile':");
        int i = 1;
        do {
            System.out.println("iteracion: " + i);
            i = i + 1;
        } while(i <= 3);
    }


    public void forLoop() {
        for (int i = 2; i < 500 && i%1==0; i = i*i) {
            System.out.println(i/2);
        }

        int j = 12;
        for (j = 2; !(j > 30); j = j*j) {
            System.out.println("El nuevo valor es: " + j);
        }

        int k = 2;
        for (; k < 10; k = k*k) {}
    }

    public void switchExample() {
        int day = 5;
        int day2 = 2;
        String dayString;
        String dayString2;
        switch (day) {
            case 1:
                switch (day) {
                    case 1:
                        dayString = "Lunes";
                        break;
                    case 2:
                        dayString = "2";
                        break;
                    default:
                        dayString = "3";
                        break;
                }
                break;
            case 2:
                dayString = "Martes";
                break;
            case 3:
                dayString = "Miercoles";
                break;
            case 4:
                dayString = "Jueves";
                break;
            case 5:
                dayString = "Viernes";
                break;
            case 6:
                dayString = "Sabado";
                break;
            case 7:
                dayString = "Domingo";
                break;
            default:
                dayString = "Dia inválido";
                break;
        }
        switch (day2) {
            case 1:
                dayString2 = "1";
                break;
            case 2:
                dayString2 = "2";
                break;
            default:
                dayString2 = "3";
                break;
        }
        System.out.println(dayString);
        System.out.println(dayString2);
    }

    public void arrayExample(){
        char arrayCaracteres[];
        arrayCaracteres = new char[10];
        arrayCaracteres[2] = 'b';
        char x = arrayCaracteres[2];

        int matriz[][];
        matriz = new int[2][2];

        matriz[1][1] = 2;
        int y = matriz[1][1];

        char arrayChar[] = {'a','b','c','d','e'};

        int arrayInt[][] = { {1,2,3,4}, {5,6,7,8}};

        System.out.println(x);
        System.out.println(y);
        System.out.println(arrayChar);
        System.out.println(arrayInt);

    }

    public void tryStatement() {
        try {
            int numerador,  denominador,resultado;
            numerador=10;
            denominador = 0;
            resultado=numerador/denominador;
            System.out.println(resultado);
        } catch (ArithmeticException ae) {
            System.out.println("No se puede dividir por cero");
        } finally {
            System.out.println("Proceso finalizado");
        }
    }

    public static void main(String[] args) {
        String demo = "Hello: " + 1 + ", World: " + 2;
        System.out.println(demo);
        JavaSourceSample j = new JavaSourceSample();
        j.show(3, "algo", args);
        System.out.println("------------------------------");
        j.forLoop();
        System.out.println("..............................");
        j.instanceFun();
        System.out.println("------------------------------");
        j.whileLoop();
        System.out.println("..............................");
        staticFun();
        System.out.println("------------------------------");
        j.switchExample();
        System.out.println("..............................");
        j.ifStatement();
        System.out.println("------------------------------");
        j.doWhileLoop();
        System.out.println("..............................");
        read();
        System.out.println("------------------------------");
        j.tryStatement();
        System.out.println("------------------------------");
        j.arrayExample();
    }
}